nautobot/ipam/models.py

```diff
@@ -612,18 +618,32 @@ def delete(self, *args, **kwargs):
             protected_objects.update(parent=self.parent)
             return super().delete(*args, **kwargs)
 
+    def get_parent(self):
+        # Determine if a parent exists and set it to the closest ancestor by `prefix_length`.
+        if self._parent is not None:
+            return self._parent
+
+        if supernets := self.supernets():
+            parent = max(supernets, key=operator.attrgetter("prefix_length"))
+            self._parent = parent
+            return parent
+        return None
+
+    def clean(self):
+        super().clean()
+
+        if self.prefix:
+            # self.parent depends on self.prefix having a value
+            self.parent = self.get_parent()
```

Comment:
Is the order here correct/desirable or do we need to set self.parent before calling super().clean()?